web/src/api/auth/login-callback.ts

```diff
@@ -113,6 +124,35 @@ export const auth0Login = withApiAuthRequired(
       }
     }
 
+    const shouldUpdateUserName =
+      auth0User.name && (!user?.name || user?.name !== auth0User.name);
```

Comment:
```suggestion
      auth0User.name && user?.name !== auth0User.name;
```
let's use Auth0 as the source of truth, same for email